From a837686416b8a70d4e553c4885e2f4250fba3325 Mon Sep 17 00:00:00 2001
From: Yongzhi Chen <ychena@apache.org>
Date: Wed, 29 Mar 2017 13:08:21 -0400
Subject: [PATCH 1072/1164] CDH-52465: HIVE-16313: Flaky test:
 TestBeeLineDriver[drop_with_concurrency] (Peter
 Vary via Yongzhi Chen)

(cherry picked from commit b176d072ed2ed33cd53f7e5189d93650246714d6)

Change-Id: I809690c80c3d8b9899291489bed31af7fecbb582
---
 .../java/org/apache/hive/beeline/qfile/QFile.java  |   23 +++++++++++++++-----
 1 file changed, 17 insertions(+), 6 deletions(-)

diff --git a/itests/util/src/main/java/org/apache/hive/beeline/qfile/QFile.java b/itests/util/src/main/java/org/apache/hive/beeline/qfile/QFile.java
index 890a167..ae5a349 100644
--- a/itests/util/src/main/java/org/apache/hive/beeline/qfile/QFile.java
+++ b/itests/util/src/main/java/org/apache/hive/beeline/qfile/QFile.java
@@ -170,17 +170,27 @@ private static String getQuotedString(File file) {
     return Shell.WINDOWS ? String.format("\"%s\"", file.getAbsolutePath()) : file.getAbsolutePath();
   }
 
+  private static class Filter {
+    private final Pattern pattern;
+    private final String replacement;
+
+    public Filter(Pattern pattern, String replacement) {
+      this.pattern = pattern;
+      this.replacement = replacement;
+    }
+  }
+
   private static class RegexFilterSet {
-    private final Map<Pattern, String> regexFilters = new LinkedHashMap<Pattern, String>();
+    private final List<Filter> regexFilters = new ArrayList<Filter>();
 
     public RegexFilterSet addFilter(String regex, String replacement) {
-      regexFilters.put(Pattern.compile(regex), replacement);
+      regexFilters.add(new Filter(Pattern.compile(regex), replacement));
       return this;
     }
 
     public String filter(String input) {
-      for (Pattern pattern : regexFilters.keySet()) {
-        input = pattern.matcher(input).replaceAll(regexFilters.get(pattern));
+      for (Filter filter : regexFilters) {
+        input = filter.pattern.matcher(input).replaceAll(filter.replacement);
       }
       return input;
     }
@@ -209,8 +219,9 @@ private static RegexFilterSet getStaticFilterSet() {
 
     return new RegexFilterSet()
         .addFilter(logPattern, "")
-        .addFilter("(?s)\nWaiting to acquire compile lock:.*?Acquired the compile lock.\n",
-            "\nAcquired the compile lock.\n")
+        .addFilter("(?s)\n[^\n]*Waiting to acquire compile lock.*?Acquired the compile lock.\n",
+            "\n")
+        .addFilter("Acquired the compile lock.\n","")
         .addFilter("Getting log thread is interrupted, since query is done!\n", "")
         .addFilter("going to print operations logs\n", "")
         .addFilter("printed operations logs\n", "")
-- 
1.7.9.5

